### DISCUSSION TITLE
"Synctex" feature in mystmd

### DISCUSSION THREAD

**Discussion by jgalazm** [executablebooks/meta#1066] [Ideas]
Hi there,

I am more of a latex user than a markdown user and though I really like mystmd what pulls me back from fully adopting it in my workflow is the "synctex" feature that most latex editors have. From the documentation I figure that, maybe, the experience fits more that of a frontend developer where you manually jump from code to live rendering of the app, focusing on one feature at a time. As a researcher who does a good amount of math in their documents, instead, I would like to be able to automatically/easily jump from a markdown line or equation to the rendered document and vice versa. This makes the writing much easier and productive than just guessing or re-reading the document in rendered or source code format. This is the "synctex" feature that you can find for example in the vscode "Latex Workshop" extension. On my case I normally cmd+click on the pdf preview which puts the cursor in the corresponding line in the editor or press cmd+option+J from the editor to scrolls the preview to the corresponding line with a visual cue.

Does this feature exist or is it planned in the near future? I think that if you could add it to the getting started guides it would be a great addition for latex users like me.

ps: I added this as a comment response in another discussion and am reposting here

**Comment by rowanc1** [comment]
Thanks @jgalazm -- this feature does not exist at the moment. However, we do have all of the information from parsing to enable it, and we also have an active websocket to the command line to route commands back -- so in principle it isn't too difficult to get going. This will depend on the editor that you use to write, what do you use for writing (e.g. vscode)?

**Reply by jgalazm** [reply]
Hi @rowanc1, I use vscode with the Latex Workshop extension.
